source/common/quic/envoy_quic_stream.h

```diff
@@ -70,22 +73,15 @@ class EnvoyQuicStream : public virtual Http::StreamEncoder,
       }
     }
 
-    if (!status_changed || *unblock_posted_) {
+    if (!status_changed) {
       return;
     }
-    // If this is the first time blocking or unblocking stream is desired, post a
-    // callback to do it in the end of the event loop. This is because unblocking a quic
-    // stream can trigger another decoding inside the callstack which messes up stream state, and
-    // blocking a quic stream in a QUICHE callstack is not expected by QUICHE.
-    *unblock_posted_ = true;
-    connection()->dispatcher().post(
-        [this, still_alive_guard = std::weak_ptr<bool>(unblock_posted_)] {
-          if (still_alive_guard.expired()) {
-            return;
-          }
-          *unblock_posted_ = false;
-          switchStreamBlockState();
-        });
+
+    // If the status transiently changed from unblocked to blocked and then unblocked, the quic
+    // stream will be spuriously unblocked and call OnDataAvailable(). This call shouldn't take any
+    // effect because any available data should have been processed already upon arrival or they
+    // were blocked by some condition other than flow control, i.e. Qpack decoding.
+    async_stream_blockage_change_->scheduleCallbackCurrentIteration();
```

Comment:
Please use scheduleCallbackNextIteration

Use of scheduleCallbackCurrentIteration can result in an infinite chain of callbacks which starve every other connection in the proxy.

Reply:
done